andela/wg-rogue-one
# #157731622: Export User workout in JSON format

### What does this PR do?
Enable export of a user's workout schedule in JSON format
#### Description of Task to be completed?
Previously, wger only supports transferring a user's workout schedule in PDF format which cannot be used by other applications.

This PR allows a user's workout information to be transferable by having their data exported in JSON format that supports interoperability.

#### Screenshots
<img width="1440" alt="screen shot 2018-06-17 at 13 23 40" src="https://user-images.githubusercontent.com/17830204/41507045-b3bf0e2c-7232-11e8-86c7-ff2df645475b.png">
The imported workouts link in navigation

<img width="1440" alt="admin_workouts" src="https://user-images.githubusercontent.com/17830204/41767996-c1c85188-7613-11e8-83d6-e6c3f97c65db.png">
The user who wants to export file

<img width="1440" alt="export" src="https://user-images.githubusercontent.com/17830204/41768022-d31347c2-7613-11e8-9e64-e2e4e03794b7.png">
The modal window for export of json files

<img width="1440" alt="imported_workouts" src="https://user-images.githubusercontent.com/17830204/41768056-e733b160-7613-11e8-8c9f-1db105fbb89c.png">
The view for imported json files

<img width="957" alt="screen shot 2018-06-16 at 04 19 51" src="https://user-images.githubusercontent.com/17830204/41494659-f3ab965a-711e-11e8-8405-d62b8358516f.png">
The exported file

<img width="1440" alt="workouts" src="https://user-images.githubusercontent.com/17830204/41768124-0cecc9fa-7614-11e8-94c8-f58dae32a364.png">
A workout that has been adopted by a user who imported a json workout

#### How should this be manually tested?
1. Follow the readme.rst procedures on how to set up the application locally 
    NB: Make sure you run the migrations command
2. Login into the system
3. From the navigation bar select workouts or Imported workouts to go to the imports view directly.
4. Select your workout.
5. On the right-hand side, open the options menu and select export/import  JSON
6. Select a tab depending on desired action. If export the file will be downloaded to your /tmp/ folder where you can move it to another location
7. On importing and exporting, you will automatically be redirected to the import and export view


#157731622
